### PR TITLE
Chore: Remove hardcoded alpine version

### PR DESCRIPTION
**Pull request by ying-jeanne**
#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by andyasp** [comment]
This introduces complexity that doesn't seem necessary.

Can we drop the patch version of Alpine instead (as in use `alpine:3.18`)? Then we'd be down to 2 updates a year, which seems easy to manage (see https://alpinelinux.org/releases/):
> There are several release branches for Alpine Linux available at the same time. Each May and November we make a release branch from edge.

As background, the alpine tags are currently uniform, so you can actually get away with a one-liner to do something like this with:
```bash
find . -name vendor -prune -o -name Dockerfile -print -o -name dev.dockerfile -print | xargs -I{} gsed -i "s/alpine:3.18.0/alpine:3.18/g" {}
```

I do think a templating approach has merits, but when I've seen it elsewhere it was to generate whole Dockerfiles with base image variants.

**Comment by charleskorn** [comment]
Adding to what @andyasp said, this change also means Dependabot won't raise PRs to automatically update the Alpine version, which seems like a step away from the goal of quickly and easily updating when we need to.

File: development/mimir-microservices-mode/compose-up.sh [hunk 1]
```diff
@@ -7,12 +7,13 @@
 set -e
 
 SCRIPT_DIR=$(cd "$(dirname "$0")" && pwd)
+ALPINE_VERSION=$(make -s -f ../../Makefile print-alpine-version)
```
Comment (pstibrany) [inline development/mimir-microservices-mode/compose-up.sh:10]:
```suggestion
ALPINE_VERSION=$(make -s -f ${SCRIPT_DIR}/../../Makefile print-alpine-version)
```

Same for other `compose-up.sh` scripts.

**Comment by pstibrany** [comment]
> Adding to what @andyasp said, this change also means Dependabot won't raise PRs to automatically update the Alpine version, which seems like a step away from the goal of quickly and easily updating when we need to.

This is a good point. If dependabot will handle Alpine updates for us automatically, why do the change in this PR in the first place? I'm happy to drop the PR and rely on dependabot.

**Comment by ying-jeanne** [comment]
Hi All, @andyasp @charleskorn @pstibrany As we have decided not to perform Dependabot updates on our Docker images at this time. As part of our weekly discussions, we have agreed to simplify the process of updating the Go/Alpine version manually to automate future updates (origin idea was just a bash script to sed whatever is needed). The next step in this pull request is to automate the entire update process by making changes to the Makefile, tagging it for releasing images. If the pull request is merged and changes the Go/Alpine version detected, it will trigger the building of the corresponding images automatically.

I would like to keep version. By explicitly specifying the patch version in the dockerfile, we ensure a consistent and reproducible environment across different deployments. This allows us to have precise control over the version of the base image used, including any specific fixes provided in the patch-level updates.

Additionally, it is essential to consider compatibility when updating to newer patch versions. There is a possibility of encountering compatibility issues when upgrading the base image to a newer patch version. These newer patches may introduce changes that could potentially affect the functionality of Mimir, its dependencies, or any libraries utilized. 

In summary, the goal of this pull request is to simplify the update process by centralizing the Alpine/Go version dependency to a single location. If you have alternative ideas to achieve this objective, please share them.

**Comment by pstibrany** [comment]
> we have decided not to perform Dependabot updates on our Docker images at this time.

Do you remember reasons for this? I assume it's because we also need to rebuild the images, and just merging PR with image version update isn't enough. But updating version via Makefile has the same problem, right?

**Comment by ying-jeanne** [comment]
> 
What do you think of utilising dependabot to automatically update all dockerfiles. Once dependabot generates PRs with the updates, we can follow a workflow where a manual tag is added to the PR, indicating that a release is needed. This tag added and PR approved would be a signal to trigger the build/push of build-image. After build-image updated in PR by bot, we can initiate the image rebuild process and merge the PR. While the rebuilding of images can be automated, it is important to have a human validation and approval step in between. This allows us to address any potential compatibility issues and ensure that the changes are thoroughly reviewed.

Using a Makefile as part of the process signifies that the rebuilding is intentionally triggered by a human action, making everyone involved more aware of the potential consequences than using the bot. But I see the convenience and ease of use provided by dependabot.

**Comment by ying-jeanne** [comment]
Based on our discussions, we have agreed upon the following approach:

- Dependabot for Alpine updates: We will utilize Dependabot to handle the automatic updates of the Alpine base image. This ensures that we stay up to date with the latest Alpine version, benefiting from its security patches and improvements. 

- Single point of truth for Go version: The single point of truth for the Go version will be maintained in mimir-build-image. This means that mimir-build-image will serve as the reference for the Go version used in our project.

- No single point of truth for Alpine: Unlike the Go version, we won't enforce a single point of truth for the Alpine base image. 

- Image rebuild triggered when mimir-build-image Dockerfile changed: The process of rebuilding images will be triggered when dockerfile changed and PR approved. Upon triggering the rebuild, the associated Dockerfile will be tagged with a unique hash. and append a commit with new build-image into PR.
